crates/weaver_semconv/src/v2/metric.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The new way we want to define metrics going forward.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    group::{GroupSpec, InstrumentSpec},
+    v2::{attribute::AttributeRef, CommonFields},
+};
+
+/// A MetricGroup defines a new metric.
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct MetricGroup {
+    /// The name of the metric.
+    pub name: String,
+    /// The instrument type that should be used to record the metric. Note that
+    /// the semantic conventions must be written using the names of the
+    /// synchronous instrument types (counter, gauge, updowncounter and
+    /// histogram).
+    /// For more details: [Metrics semantic conventions - Instrument types](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions#instrument-types).
+    /// Note: This field is required if type is metric.
+    pub instrument: InstrumentSpec,
+    /// The unit in which the metric is measured, which should adhere to the
+    /// [guidelines](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions#instrument-units).
+    pub unit: String,
```

Comment:
[comment] If one day, someone has time to make a unit parser, that would be cool!